doc/contribute/external.rst

```diff
@@ -103,6 +103,38 @@ With this approach the code is considered as being developed externally, and
 thus it is not automatically subject to the requirements of the previous
 section.
 
+Integration in main manifest file (west.yaml)
++++++++++++++++++++++++++++++++++++++++++++++
+
+Integrating external code into the main :file:`west.yml` manifest file is
+limited to code that is used by a Zephyr subsystem (libraries), by a platform,
+drivers (HAL) or tooling needed to test or build Zephyr components.
+
+The integration of modules in this group is validated by the Zephyr project CI,
+and verified to be working with each Zephyr release.
+
+Integration as optional modules
++++++++++++++++++++++++++++++++
+
+Standalone or loose integration of modules/projects without any incoming
+dependencies shall be made optional and shall be kept standalone. Optional
+projects that provide value to users directly and through a Zephyr subsystem or
+platform shall be added to an optional manifest file that is filtered by
+default. (:file:`submanifests/optional.yml`).
+
+Such optional projects might include samples and tests in their own repositories.
+
+There shall not be any direct dependency added in the Zephyr code tree and all
+sample or test code shall be maintained within the module tree.
```

Comment:
AFAICT this is the first time "module tree" is being introduced. I'm not sure what is meant by this.  Consider the following interpretations:

The sample or test code needs to be:

1. In the module repository;
2. In directories pointed to by module.yml;
3. In directories following the Zephyr directory tree structure;
4. In directory sub-trees dedicated to the Zephyr build system;

Once we get to the Zephyr directory tree structure in a module then the build system must support that tree being rooted in the same directory as `module.yml`. This means a module usually has code that is outside the Zephyr directory tree structure (clearly "out-of-tree"). 

We don't have an unambiguous term for code located in a Zephyr directory structure in a repository other than zephyr.git. What should it be called?

Reply:
> AFAICT this is the first time "module tree" is being introduced. I'm not sure what is meant by this.

git repo? If that's all it is then "git repo" is much clearer.

West is git-only and AFAIK with no plan to support anything else any time soon.

Reply:
"git repo" can also refer to `zephyr.git`, so that's not a very good option either.

For Zephyr, west is explicitly supposed to be optional, so its terms are not good either.

To someone brand new to Zephyr, "in-tree" very reasonably means anything/everything within a Zephyr directory structure, no matter which repository or which part of a repository it is in (e.g., example_application, hal/silabs, audio/sof (partially)).

Soon we'll have to talk about {component/file} within the "{module_name} module". Perhaps there is a natural extension of that pattern to cover every case.

Reply:
> "git repo" can also refer to zephyr.git, so that's not a very good option either.

I meant "module's git repo" of course.

> For Zephyr, west is explicitly supposed to be optional, so its terms are not good either.

This PR and feature is ALL about west?!?

https://en.wikipedia.org/wiki/Sealioning